packages/grid/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx

```diff
@@ -39,7 +39,6 @@ interface GridGenericColumnHeaderItemProps
   label: string;
   draggableContainerProps?: Partial<React.HTMLProps<HTMLDivElement>>;
   columnHeaderSeparatorProps?: Partial<GridColumnHeaderSeparatorProps>;
-  disableHeaderSeparator?: boolean;
```

Comment:
@mui/xgrid Team, I couldn't find this prop being referenced or used anywhere, is it already deprecated and is waiting to be removed or there is something else to it?

Reply:
It was added in https://github.com/mui/mui-x/pull/5133 for column grouping. It's used internally since the component is not exported:
https://github.com/mui/mui-x/blob/d8b76d392976d8c1c9eeeb37d13a6d2368751f9a/packages/grid/x-data-grid/src/components/columnHeaders/GridColumnGroupHeader.tsx#L193